File: codeforces/039/J.cpp
#include <string>
#include <iostream>
#include <algorithm>

using namespace std;

int main() {
	string a, b;
	int l, r;

	cin >> a >> b;
	if (a.length() != b.length() + 1) {
		l = 0;
		r = -1;
	} else {
		r = 0;
		while (r < (int)b.length() && a[r] == b[r]) {
			++r;
		}
		l = (int)b.length() - 1;
		while (l >= 0 && a[l + 1] == b[l]) {
			--l;
		}
		cout << max(0, r - l) << endl;
		for (int i = 0; i <= (int)b.length(); ++i) {
			if (l < i && i <= r) {
				cout << i + 1 << ' ';
			}
		}
		cout << endl;
	}

	return 0;
}

//# 	When 	Who 	Problem 	Lang 	Verdict 	Time 	Memory
//402305 	Apr 19, 2011 11:24:06 PM 	watashi 	39J - Spelling Check 	GNU C++ 	Accepted 	610 ms 	4000 KB 

File: codeforces/ac/0/045/J.rb
n, m = gets.split.map(&:to_i)
if [n, m] == [1, 1]
	a = [[1]]
elsif [n, m] == [1, 4]
	a = [[2, 4, 1, 3]]
elsif [n, m] == [4, 1]
	a = [[2], [4], [1], [3]]
elsif n * m <= 4
	a = [[-1]]
elsif n == 1
	a = [[*1 .. m].partition{|_| _.odd?}.flatten]
elsif m == 1
	a = [*1 .. n].partition{|_| _.odd?}.flatten.map{|_| [_]}
else
	a = Array.new(n){Array.new(m)}
	k = 1
	n.times do |i|
		m.times do |j|
			until a[i][j]
				a[i][j] = k
				k += 1
				i = (i + 1) % n
				j = (j + 1) % m
			end
		end
	end
end
puts a.map{|_| _ * ' '} * "\n"
__END__
# 	When 	Who 	Problem 	Lang 	Verdict 	Time 	Memory
413381 	Apr 28, 2011 2:33:41 AM 	watashi 	45J - Planting Trees 	Ruby 	Accepted 	90 ms 	4200 KB
